util/util.go

```diff
@@ -149,25 +150,25 @@ func ModifyImageTag(imageName, tagName string) (string, error) {
 
 // ModifyImageRepository takes an imageName (e.g., repository/image:tag), and returns an image name with updated repository
 func ModifyImageRepository(imageName, repositoryName string) (string, error) {
```

Comment:
Shouldn't we be able to get rid of these `ModifyImage` helpers entirely? I would think you can just do like 

```
image, _ := container.ParseImageName(imageName)
image.Repository = repositoryName
image.String()
```

Reply:
if we remove all the calls to the helpers, we have to duplicate a bunch of validation,  but tried to make the use of the `docker reference` helpers in one file only